### ISSUE TITLE
How can I initialise weights in darknet? 

### ISSUE THREAD

**Issue by hardik22317** [AlexeyAB/darknet#3419]
I have built a retinanet cfg, resnext152 is the backbone that I have used, now when I am using the resnext's pretrained weights, my loss is becoming nan in the first iteration only. So, I think that there might be a problem in the weight initialisation (there could be any other too). So, how can I initialise the weights according to the weight initialisation process of Retinanet mentioned [here](https://arxiv.org/pdf/1708.02002.pdf)

**Comment by hardik22317** [comment]
I have built a retinanet cfg, resnext152 is the backbone that I have used, now when I am using the resnext's pretrained weights, my loss is becoming nan in the first iteration only. So, I think that there might be a problem in the weight initialisation (there could be any other too). So, how can I initialise the weights according to the weight initialisation process of Retinanet mentioned [here](https://arxiv.org/pdf/1708.02002.pdf)

**Comment by hardik22317** [comment]
@AlexeyAB sir, can you please help me on this issue?

**Comment by AlexeyAB** [comment]
@hardik22317 Hi,

Try to update your code (there was a bug)
and train again with GPU=1 CUDNN=1.

----

There are lines that initialize convolutional weights (if pre-trained file isn't used): https://github.com/AlexeyAB/darknet/blob/8c80ba6b53429347bd6249a16cd576c971a0776b/src/convolutional_layer.c#L381-L382

**Comment by hardik22317** [comment]
Thank you @AlexeyAB, for getting back and replying, will try this and let you know the outcome. Thank you so much.

**Comment by hardik22317** [comment]
@AlexeyAB sir, I tried what you have suggested. I followed the procedure you have mentioned [here](https://github.com/AlexeyAB/darknet#how-to-train-with-multi-gpu). But, in that also, everything was working fine, but from 3096th iteration, loss became increasing (almost double) and from 3160th iteration, it became nan. Could you help me with this?

Thank you.

**Comment by hardik22317** [comment]
@AlexeyAB I am training Retinanet on COCO dataset, but at 624th iteration, I get avg loss to be nan. I am attaching my cfg below. 
[resnext152_32_4d_retinanet.txt](https://github.com/AlexeyAB/darknet/files/3308872/resnext152_32_4d_retinanet.txt)
